### DISCUSSION TITLE
Abandoned Cart Notifications

### DISCUSSION THREAD

**Discussion by andris-sevcenko** [craftcms/commerce#2308] [Ideas]
> Created by: Brendan Underwood ([bunderwood@brandagency.com.au](bunderwood@brandagency.com.au)) on 2015/12/13 06:44:54 +0000
Votes at time of UserVoice import: 25

------
This function is available on a lot of ecommerce systems like Big Commerce. The ability to send a notification out (after a time period) to a user that's added items into their cart then abandoned it. I've seen it done where sometimes they may provide an incentive too like a 10% discount.

**Comment by andris-sevcenko** [comment]
> Created by: David Ritter ([david.ritter@spinthis.com](david.ritter@spinthis.com)) on 2016/11/05 19:55:23 +0000


------
Combine this with an integration with something like MailChimp, aweber, etc. and you&#39;d have something on par with the big boys.

For example, I&#39;m not sure what it would take to get on MailChimp&#39;s Connect Your Store functionality but as an end user, showing compatibility with Craft Commerce would be a huge selling point. Otherwise I&#39;m getting swayed by Magento, etc.

For example:
https://mailchimp.com/connect-your-store/

**Comment by andris-sevcenko** [comment]
> Created by: Hagen Rode ([hagen00@gmail.com](hagen00@gmail.com)) on 2017/01/17 08:01:24 +0000


------
Very necessary! +1

**Comment by andris-sevcenko** [comment]
> Created by: Hagen Rode ([hagen00@gmail.com](hagen00@gmail.com)) on 2017/01/17 08:16:52 +0000


------
For anyone looking to build this as a plugin. Feedback from Luke (BDFL):
&quot;Shouldn&#39;t be too hard for a plugin to do. Find a cart, then send an email with a link to a plugin controller action (that includes the cart `number` as a param) and then set the cart cookie to that number.&quot;

**Comment by andris-sevcenko** [comment]
> Created by: Fred Carlsen ([fred@sjelfull.no](fred@sjelfull.no)) on 2017/03/28 10:21:05 +0000


------
Register your interest for my plugin here: https://superbig.co/plugins/abandoned-cart-recovery

**Comment by andris-sevcenko** [comment]
> Created by: Hagen Rode ([hagen00@gmail.com](hagen00@gmail.com)) on 2017/06/21 12:38:42 +0000


------
That&#39;s great Fred, I&#39;m glad to see that it&#39;s being actively maintained as well. Will definitely be purchasing!

**Comment by andris-sevcenko** [comment]
> Created by: Digital Agency WHITE ([contact@white.nl](contact@white.nl)) on 2017/09/01 09:43:28 +0000


------
We created this as a simple plugin. You can find it here:
https://straightupcraft.com/craft-plugins/abandoned-cart

**Comment by sjcallender** [comment]
You'll typically want to use a third-party service for managing marketing emails. We've built Klaviyo Connect so you can Klaviyo to manage all abandoned cart (and other marketing) email series. https://github.com/FosterCommerce/klaviyoconnect

**Comment by lukeholder** [comment]
See: https://plugins.craftcms.com/abandoned-cart

**Comment by lukeholder** [comment]
Commerce 3.1 now supports a special URL that can recover a cart #1348

**Comment by terryupton** [comment]
Is this planned to become part of the Craft commerce core at any point in the future?

**Reply by sjcallender** [reply]
@terryupton - I wouldn't count on it. Also, IMO, it's best handled with a service that set rules for when to send the email, what content to send, and who specifically (segmenting) to send it to. Not to mention the ability to track related sales originated from links from the emails.

**Reply by terryupton** [reply]
Thanks @sjcallender - I've actually just been looking at your Klaviyo Connect plugin and how this might work for this and beyond. 
looks interesting and I am going to follow up and suggest this to the client. 
Will you be looking to add Commerce 4 support for this? I am considering implementing something for a client tin their existing commerce 3 setup, but I don't want this to become a future blocker to Commerce 4.
